### PR TITLE
chore: add `.pre-commit-hooks.yaml`

### PR DESCRIPTION
**Pull request by 34j**
This PR adds a [pre-commit](https://pre-commit.com/#new-hooks) hook.

### CODE REVIEW

**Comment by Enter-tainer** [comment]
hi, can you briefly explain what is this used for?

**Comment by 34j** [comment]
I can't explain it well, but pre-commit is the most famous CLI tool for managing Git Hooks, next to Husky.

**Comment by Enter-tainer** [comment]
> The hook must exit nonzero on failure or modify files.

I think typstyle doesn't support this at this moment?

File: README.md [hunk 1]
```diff
@@ -71,6 +71,17 @@ typstyle has been integrated into [tinymist](https://github.com/Myriad-Dreamin/t
 
 There is an online version of the formatter at <https://enter-tainer.github.io/typstyle/> that you can see how it formats your code.
 
+### Use with [pre-commit](https://github.com/pre-commit/pre-commit)
+
+Add this to your `.pre-commit-config.yaml`:
+
+```yaml
+  - repo: https://github.com/Enter-tainer/typstyle
+    rev: ''  # Use the sha / tag you want to point at
```
Comment (Enter-tainer) [inline README.md:80]:
```suggestion
    rev: ''  # The the revision or tag you want to use
```

**Comment by 34j** [comment]
If typstyle does not do linting, then I think the current behavior is fine.

**Comment by Enter-tainer** [comment]
Sorry but I don't really understand what pre-commit does. I guess, it will format files before you commit them? Or it will refuse to do `git commit` because you have unformatted files?

**Comment by 34j** [comment]
- Runs hooks before committing. 
- If an error occurs (linting / formatting) or a file is changed (formatting), the commit is not performed. 
- The files on the machine are formatted and you can see what has changed by comparing them to the staged files.

**Comment by Enter-tainer** [comment]
> or a file is changed (formatting), the commit is not performed.

Ah.. I think typstyle will fall into this path then? And if your typ source is not formatted, it will block you from git-commiting.

**Comment by Enter-tainer** [comment]
Thank you!

**Pull request merged**